strangerstudios/pmpro-roles
# Default role removal after level change

### All Submissions:


### Changes proposed in this Pull Request:

Resolves the issue of old role staying after membership checkout.

These changes adds the default role to the $old_roles array instead of using `$user->remove_role( $default_role );` further down.

Setting up the default role to be removed when the user has 2 roles and the default role is one of the two, has the unintended effect of removing the default role if the user assigns the default role plus one other role to a level.

This also fixes the issue of default role not being removed when it is not an assigned role in any of the user's old membership levels, or when the user starts off with no membership levels at all.

### How to test the changes in this Pull Request:

1. Assign one role (not default role) to a membership level
2. Sign up a new member to that level (notice the default role is removed)
3. Assign default role plus one other role to a membership level
4. Sign up a new member to that level (notice the default role is removed)
5. Assign two roles (not default role) to a membership level
6. Sign up a new member to that level (notice the default role is NOT removed)
7. Apply changes in this pull request and try 1-6 again
8. default role would NOT be removed when default role plus one other role is assigned to a membership level
9. default role would be removed when two roles (neither role default) are assigned to a membership level

### Other information:


### Changelog entry

BUG FIX: When default role plus one other role is assigned to a membership level, default role is no longer removed. When two roles or more are assigned to a membership level, and none of those are default roles, the default role would now be removed.
